tests_e2e/tests/extensions_policy_disabled/extensions_policy_disabled.py

```diff
@@ -0,0 +1,82 @@
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+#
+# This test disables extension processing on waagent.conf and verifies that extensions are not processed, but the
+# agent continues reporting status.
+#
+
+import uuid
+
+from assertpy import assert_that, fail
+from typing import Any
+
+from tests_e2e.tests.lib.agent_test import AgentVmTest
+from tests_e2e.tests.lib.vm_extension_identifier import VmExtensionIds
+from tests_e2e.tests.lib.logging import log
+from tests_e2e.tests.lib.ssh_client import SshClient
+from tests_e2e.tests.lib.virtual_machine_extension_client import VirtualMachineExtensionClient
+
+
+class ExtensionsPolicyDisabled(AgentVmTest):
+    class TestCase:
+        def __init__(self, extension: VirtualMachineExtensionClient, settings: Any):
+            self.extension = extension
+            self.settings = settings
+    def run(self):
+        ssh_client: SshClient = self._context.create_ssh_client()
+
+        # Enable policy engine on the test VM
+        log.info("")
+        log.info("Enable extension policy enforcement on the test VM [%s]", self._context.vm.name)
+        output = ssh_client.run_command("update-waagent-conf Debug.EnableExtensionPolicy=y", use_sudo=True)
+        log.info("Enable policy completed:\n%s", output)
+
+        # Prepare test cases
+        # TO DO - add more extensions/cases after full policy functionality is implemented
+        unique = str(uuid.uuid4())
+        test_file = f"waagent-test.{unique}"
+        test_cases = [
+            ExtensionsPolicyDisabled.TestCase(
+                VirtualMachineExtensionClient(self._context.vm, VmExtensionIds.CustomScript,
+                                              resource_name="CustomScript"),
+                {'commandToExecute': f"echo '{unique}' > /tmp/{test_file}"}
+            )
+        ]
+
+        for t in test_cases:
+            log.info("")
+            log.info("Test case: %s", t.extension)
+
+            log.info("Policy engine initialization should fail.")
+            try:
+                t.extension.enable(settings=t.settings, force_update=True, timeout=6 * 60)
+                log.info("Checking that policy engine initialization is skipped...")
+                expected_msg = "Extension policy is not enabled. Continuing without policy enforcement."
+                ssh_client.run_command("grep \"{0}\" /var/log/waagent.log".format(expected_msg))
+                log.info("Skipped policy enforcement as expected.")
+            except Exception as error:
+                fail(f"Unexpected error while processing {t.extension.__str__()}")
```

Comment:
we need to catch the exception raised by run_command when grep doesn't find the string and report that as a test error, rather than as an unexpected error

Reply:
same comment for other test